src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -158,10 +162,13 @@ object ScalafixPlugin extends AutoPlugin {
         SettingKey[Boolean]("bspEnabled") := false
       )
     ),
+    scalafixCallback := new ScalafixLogger(ScalafixInterface.defaultLogger),
     scalafixInterfaceProvider := ScalafixInterface.fromToolClasspath(
       scalafixScalaBinaryVersion.value,
       scalafixDependencies = scalafixDependencies.value,
-      scalafixCustomResolvers = scalafixResolvers.value
+      scalafixCustomResolvers = scalafixResolvers.value,
+      logger = ScalafixInterface.defaultLogger,
+      callback = scalafixCallback.value
```

Comment:
Watch out for:
https://github.com/scalacenter/sbt-scalafix/blob/c2777cfe1ea5623201fc3ae86d74e1043e19f4f3/src/main/scala/scalafix/internal/sbt/ScalafixInterface.scala#L131-L136

As it stands in the current state of this PR, a client might set different values for `scalafixCallback` across projects, and see only one being used because of memoization.

We can either
1. add `callback` as a cache key (using reference equality I guess?)
1. support only one value per build
```suggestion
      callback = (ThisBuild / scalafixCallback).value
```

I would go for (1) for consistency with other configurable attributes of `scalafixInterface`

Reply:
Reopening this for visibility: given the current state of sbt-scalafix, there is a small limitation to be aware of in the solution (2) you went for. The [cache is never flushed](https://github.com/scalacenter/sbt-scalafix/blob/a540aebe45b62f67530d11b484f33b3a46624784/src/main/scala/scalafix/internal/sbt/ScalafixInterface.scala#L119-L122), so if a sbt command changed the state of the build and update `ThisBuild / scalafixCallback`, you will get false positive cache hits.

This is a corner case, but depending on how you end up injecting a per-invocation `scalafixCallback` to avoid mutability at that level, this could be annoying. If that's the case, we'll have to go for (1), as well as maybe keep the cache in a setting key to avoid leaking memory upon each `sbt reload` because of a new JVM instance of `ScalafixMainCallback`.